contracts/NonFungibleCollectable.sol

```diff
@@ -0,0 +1,42 @@
+pragma solidity ^0.4.18;
+
+import './lib/NonFungibleToken.sol';
+import './lib/Ownable.sol';
+
+contract NonFungibleCollectable is NonFungibleToken, Ownable {
+  address public collectableCouponIssuer;
+
+  function NonFungibleCollectable(string name, string symbol, address couponIssuer)
+    public
+    NonFungibleToken(name, symbol)
+  {
+    collectableCouponIssuer = couponIssuer;
+  }
+
+
+  function ownerMint(address _to, uint256 _tokenId)
+    public
+    onlyOwner
+  {
+    super.mint(_to, _tokenId);
+  }
+
+  function hasher(uint tokenId) public view returns(bytes32) {
+    return keccak256(tokenId, msg.sender);
+  }
+
+  function getAddress(bytes32 hash, uint8 v, bytes32 r, bytes32 s) public pure returns(address) {
+    bytes memory prefix = "\x19Ethereum Signed Message:\n32";
```

Comment:
Would you mind leaving a few comments in this file? In particular, could you explain the need to have `prefix` in `memory` rather than `storage` as well as the form of the `prefix` you have gone with here? What exactly is going on there?

Reply:
Very good point! Haha, this should be done off chain! :)

Reply:
👍 cool, was thinking that there was something else weird and new I'd need to learn about `:/n32` and stuff 😅

Reply:
Hmm, very wierd the hash I'm getting by computing this off chain is different, I'll get it right though, I hope :sweat_smile: